control/plugin/client/client.go

```diff
@@ -41,6 +43,17 @@ type PluginCollectorClient interface {
 	GetMetricTypes(plugin.ConfigType) ([]core.Metric, error)
 }
 
+type PluginStreamCollectorClient interface {
+	PluginClient
+	StreamMetrics([]core.Metric) (chan []core.Metric, chan error, error)
+	GetMetricTypes(plugin.ConfigType) ([]core.Metric, error)
+	UpdateCollectedMetrics([]core.Metric) error
+	UpdatePluginConfig([]byte) error
+	UpdateMetricsBuffer(int64) error
+	UpdateCollectDuration(time.Duration) error
+	Killed()
```

Comment:
The grpc kill method isn't called in all cases. Under some circumstances we will kill the plugin process directly and want to ensure that we notify the client/scheduler that it is happening.